app/assets/javascripts/Components/Result/result.jsx

```diff
@@ -769,13 +825,32 @@ class Result extends React.Component {
     });
   };
 
+  refreshFilterData = () => {
+    const storedFilter = localStorage.getItem(
+      `${this.props.user_id}_${this.state.assignment_id}_filterData`
+    );
+    if (storedFilter) {
+      this.setState({filterData: JSON.parse(storedFilter)});
+    } else {
+      this.setState({filterData: INITIAL_FILTER_MODAL_STATE});
```

Comment:
Reaplace this bare `setState` call with a call to `updateFilterData` so that the localStorage also gets reset.